redisenterprise/datadog_checks/redisenterprise/check.py

```diff
@@ -0,0 +1,320 @@
+            )
+            self.gauge(
+                'redisenterprise.memory_limit',
+                bdb_dict[int(i)]['limit'],
+                tags=tgs + service_check_tags,
+                hostname=host,
+            )
+            # derive our own stats from others
+            self.gauge(
+                'redisenterprise.used_memory_percent',
+                100 * stats[i]['used_memory'] / bdb_dict[int(i)]['limit'],
+                tags=tgs + service_check_tags,
+                hostname=host,
+            )
+            # derive our cache hit rate - be sure not to divide by 0
+            if (
+                stats[i]['read_hits'] + stats[i]['read_misses'] + stats[i]['write_hits'] + stats[i]['write_misses']
+            ) == 0:
+                self.gauge(
+                    'redisenterprise.cache_hit_rate',
+                    0.0,
+                    tags=tgs + service_check_tags,
+                    hostname=host,
+                )
+            else:
+                self.gauge(
+                    'redisenterprise.cache_hit_rate',
+                    100
+                    * (stats[i]['read_hits'] + stats[i]['write_hits'])
+                    / (
+                        stats[i]['read_hits']
+                        + stats[i]['read_misses']
+                        + stats[i]['write_hits']
+                        + stats[i]['write_misses']
+                    ),
+                    tags=tgs + service_check_tags,
+                    hostname=host,
+                )
+            # derive flash object percentage being sure that the key esists and is not 0
+            if 'bigstore_objs_flash' in stats[i].keys():
+                if stats[i]['bigstore_objs_flash'] > 0:
+                    self.gauge(
+                        'redisenterprise.bigstore_objs_percent',
+                        100
+                        * stats[i]['bigstore_objs_ram']
+                        / (stats[i]['bigstore_objs_ram'] + stats[i]['bigstore_objs_flash']),
+                        tags=tgs + service_check_tags,
+                        hostname=host,
+                    )
+
+            for j in stats[i].keys():
+                if j in gauges:
+                    self.gauge(
+                        'redisenterprise.{}'.format(j), stats[i][j], tags=tgs + service_check_tags, hostname=host
+                    )
+        return 0
+
+    def _get_license(self, host, port, timeout, auth, verifyssl, service_check_tags):
+        """ Collect Enterprise License Information """
+        stats = self._api_fetch_json(host, port, timeout, auth, verifyssl, "license", service_check_tags)
+        expire = datetime.strptime(stats['expiration_date'], "%Y-%m-%dT%H:%M:%SZ")
+        now = datetime.now()
+        self.gauge('redisenterprise.license_days', (expire - now).days, tags=service_check_tags, hostname=host)
+        self.gauge('redisenterprise.license_shards', stats['shards_limit'], tags=service_check_tags, hostname=host)
```

Comment:
```suggestion
        self.gauge('redisenterprise.license_days', (expire - now).days, tags=service_check_tags)
        self.gauge('redisenterprise.license_shards', stats['shards_limit'], tags=service_check_tags)
```
hostname is meant to be the agent host and not the check host. This variable is autofilled by the agent and should not be used except in very specific occasions